src/Data/LLVM/BitCode/BitString.hs

```diff
@@ -13,26 +19,49 @@ import Numeric (showIntAtBase)
 
 import Prelude hiding (take,drop,splitAt)
 
+newtype NumBits = NumBits Int deriving (Show, Eq, Ord)
+newtype NumBytes = NumBytes Int deriving (Show, Eq, Ord)
+
+pattern Bits' :: Int -> NumBits
+pattern Bits' n = NumBits n
+{-# COMPLETE Bits' #-}
+
+pattern Bytes' :: Int -> NumBytes
+pattern Bytes' n = NumBytes n
+{-# COMPLETE Bytes' #-}
+
+bitCount :: NumBits -> Int
+bitCount (NumBits n) = n
+
+addBitCounts :: NumBits -> NumBits -> NumBits
+addBitCounts (NumBits a) (NumBits b) = NumBits $ a + b
+
+subtractBitCounts :: NumBits -> NumBits -> NumBits
+subtractBitCounts (NumBits a) (NumBits b) = NumBits $ a - b
```

Comment:
These look like the beginnings of a `Num` instance. Would it be easier just to define one and use that rather than these assorted top-level functions? (A similar question for `NumBytes` as well.)

Reply:
These are actually going away in an upcoming PR.  It's a good suggestion if I were keeping them, but the process here was  to shift from standard classes to explicit functions to track the usage preparatory to more substantial changes.

Reply:
Gotcha.